Library/Homebrew/formula.rb

```diff
@@ -1249,9 +1249,19 @@ def fish_function = share/"fish/vendor_functions.d"
   # This is symlinked into `HOMEBREW_PREFIX` after installation or with
   # `brew link` for formulae that are not keg-only.
   #
+  # By default the compatibility directory is used to support system Bash 3
+  # on macOS. Pass in `compat: false` when installing completions that need
+  # Bash >= 4 or bash-completion >= 2.
+  #
   # @api public
-  sig { returns(Pathname) }
-  def bash_completion = prefix/"etc/bash_completion.d"
+  sig { params(compat: T::Boolean).returns(Pathname) }
+  def bash_completion(compat: true)
```

Comment:
```suggestion
  def bash_completion(bash3: false)
```
maybe?

Reply:
I had went with `compat` due to bash-completion naming, though I'm open to alternatives .
```console
❯ pkgconf bash-completion --variable compatdir --define-prefix
/opt/homebrew/etc/bash_completion.d

❯ pkgconf bash-completion --variable completionsdir --define-prefix
/opt/homebrew/share/bash-completion/completions
```

Some other ideas I was considering:
- `bash_completion(use_compatdir: false)`
- `bash_completion(version = :v1)`, so `bash_completion(:v2).install ...`

---

If using `bash3`, it should also default to `true` to preserve existing behavior.